src/Payments.sol

```diff
@@ -488,37 +487,38 @@ contract Payments is
         uint256 lockupFixed
     ) internal {
         Account storage payer = accounts[rail.token][rail.from];
+        bool isIncrease = period > rail.lockupPeriod ||
+            lockupFixed > rail.lockupFixed;
 
         // Settle account lockup as much as possible
         // Only require full settlement if increasing period or fixed lockup
-        if (period > rail.lockupPeriod || lockupFixed > rail.lockupFixed) {
+        if (isIncrease) {
             require(
                 payer.lockupLastSettledAt == block.number,
                 "cannot increase lockup: client funds insufficient for current account lockup settlement"
             );
-        } else if (period < rail.lockupPeriod) {
-            // When reducing period, ensure we still cover all unsettled epochs
+        } else if (period < rail.lockupPeriod && !isRailInDebt(rail, payer)) {
```

Comment:
@Stebalien Can you eyeball this ? Basically we don't want to allow a period reduction below the current epoch if rail is already fully settled upto the current epoch.